scripts/envDFL.sh

```diff
@@ -0,0 +1,273 @@
+}
+
+export_var_env() {
+    local var="$@"
+    local name=${var%%=*}
+    local value="${var#*=}"
+
+    if ! `expr $name : "DYNAFLOW_LAUNCHER_.*" > /dev/null`; then
+      error_exit "You must export variables with DYNAFLOW_LAUNCHER_ prefix for $name."
+    fi
+
+    if eval "[ \"\$$name\" ]"; then
+      eval "value=\${$name}"
+      ##echo "Environment variable for $name already set : $value"
+      return
+    fi
+
+    if [ "$value" = UNDEFINED ]; then
+      error_exit "You must define the value of $name"
+    fi
+    export $name="$value"
+}
+
+export_var_env_force() {
+  local var="$@"
+  local name=${var%%=*}
+  local value="${var#*=}"
+
+  if ! `expr $name : "DYNAFLOW_LAUNCHER_.*" > /dev/null`; then
+    error_exit "You must export variables with DYNAFLOW_LAUNCHER prefix for $name."
+  fi
+
+  if eval "[ \"\$$name\" ]"; then
+    unset $name
+    export $name="$value"
+    return
+  fi
+
+  if [ "$value" = UNDEFINED ]; then
+    error_exit "You must define the value of $name"
+  fi
+  export $name="$value"
+}
+
+ld_library_path_remove() {
+  export LD_LIBRARY_PATH=`echo -n $LD_LIBRARY_PATH | awk -v RS=: -v ORS=: '$0 != "'$1'"' | sed 's/:$//'`;
+}
+
+ld_library_path_prepend() {
+  if [ ! -z "$LD_LIBRARY_PATH" ]; then
+    ld_library_path_remove $1
+    export LD_LIBRARY_PATH="$1:$LD_LIBRARY_PATH"
+  else
+    export LD_LIBRARY_PATH="$1"
+  fi
+}
+
+define_options() {
+    export_var_env DYNAFLOW_LAUNCHER_USAGE="Usage: `basename $0` [option] -- program to deal with DynaFlow Launcher
+
+where [option] can be:
+        =========== Build
+        build-user                              build DynaFlow Launcher
+        build-tests-coverage                    build DynaFlow Launcher and run coverage
+
+        =========== Tests
+        tests                                   launch DynaFlow Launcher unit tests
+
+        =========== Launch
+        launch-dir [dir]                        launch DynaFlow Launcher with:
+                                                - network filepath = <dir>/network.xml (only IIDM is supported)
+                                                - configuration filepath = <dir>/config.json
+        launch [network] [config]               launch DynaFlow Launcher
+
+        =========== Others
+        help                                    show all available options
+        version                                 show DynaFlow Launcher version
+        reset-environment                       reset all environment variables set by DynaFlow Launcher
+"
+}
+
+help() {
+    define_options
+    echo "$DYNAFLOW_LAUNCHER_USAGE"
+}
+
+set_commit_hook() {
+    $HERE/set_commit_hook.sh
+}
+
+set_environment() {
+    # global vars
+    ld_library_path_prepend $DYNAWO_INSTALL_DIR/lib         # For Dynawo library
+    ld_library_path_prepend $DYNAFLOW_LAUNCHER_HOME/lib64   # For local DFL libraries, used only at runtime in case we compile in shared
+
+    # dynawo vars
+    export DYNAWO_INSTALL_DIR=$DYNAWO_HOME
```

Comment:
Maybe we should test that DYNAWO_HOME exists

Reply:
DYNAWO_HOME is marked as required in the readme and it is already checked in the cmakelist